ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+TabWebDelegate.swift

```diff
@@ -447,9 +450,57 @@ extension BrowserViewController: TabWebDelegate {
     }
     return false
   }
+
+  func tab(_ tab: Tab, buildEditMenuWithBuilder builder: any UIMenuBuilder) {
+    let forcePaste = UIAction(title: Strings.forcePaste) { [weak tab] _ in
+      if let string = UIPasteboard.general.string {
+        tab?.evaluateSafeJavaScript(
+          functionName: "window.__firefox__.forcePaste",
+          args: [string, UserScriptManager.securityToken],
+          contentWorld: .defaultClient
+        ) { _, _ in }
+      }
+    }
+    let searchWithBrave = UIAction(title: Strings.searchWithBrave) { [weak tab, weak self] _ in
+      tab?.evaluateSafeJavaScript(
+        functionName: "getSelection().toString",
+        contentWorld: .defaultClient
+      ) {
+        result,
+        _ in
+        guard let tab, let selectedText = result as? String else { return }
+        self?.didSelectSearchWithBrave(selectedText, tab: tab)
+      }
+    }
+    let braveMenuItems: UIMenu = .init(
+      options: [.displayInline],
+      children: [forcePaste, searchWithBrave]
```

Comment:
Unfortunately there's no way to detect in `paste` is in the menu because the actual command is always in the menu builder, so Force Paste for the time being will have to always be visible.  In the future we can possibly pass in the invoker to manually run `canPerformAction` on the view that will handle the action to see if its appropriate to display the menu item, but we'd need to find a way to handle this in Chromium as well